modules/calib3d/src/solvepnp.cpp

```diff
@@ -357,7 +359,7 @@ void cv::solvePnPRansac(InputArray _opoints, InputArray _ipoints,
                 Mat colInlierObjectPoints = inlierObjectPoints(Rect(i, 0, 1, 1));
                 objectPoints.col(index).copyTo(colInlierObjectPoints);
             }
-            solvePnP(inlierObjectPoints, inlierImagePoints, params.camera.intrinsics, params.camera.distortion, localRvec, localTvec, true, flags);
+            solvePnP(inlierObjectPoints, inlierImagePoints, params.camera.intrinsics, params.camera.distortion, localRvec, localTvec, false, flags);
```

Comment:
In `PnPSolver::operator()`, if a sample/iteration has more inliers than the minimum inlier count (`if ((int)inliers.size() >= parameters.minInliersCount)`), then we terminate early.  Thus, if there are two samples that potentially have more inliers than `minInliersCount`, there is a race condition as to which sample will be chosen, yielding different results.

One might guess that this issue is resolved by setting minInliersCount to -1, which requires all points to be inliers in order to exit early.  In this situation, the `localInliers` vector will be the same for both samples, but the `localRvec` and `localTvec` variables will differ.  Thus, if `localRVec` and `localTvec` are then used to compute the final `solvePnP` solution, we will have a non-determinism.  Setting the `useExtrinsicGuess` option to false avoids the use of `localRvec` and `localTvec` in the final `solvePnP` solve, and thus keeps the algorithm deterministic when `minInliersCount` is `-1`.